presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -671,6 +674,10 @@ private AmazonS3 createAmazonS3Client(URI uri, Configuration hadoopConfig, Clien
             clientBuilder = clientBuilder.withRegion(US_EAST_1);
         }
 
+        if (globalBucketAccessEnabled) {
+            clientBuilder.setForceGlobalBucketAccessEnabled(true);
```

Comment:
What would be the drawbacks if this was always on (as it effectively used to be before migration to client builder)?

https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Builder.html#setForceGlobalBucketAccessEnabled-java.lang.Boolean- lists one drawback:
> Enabling this mode has several drawbacks, because it has the potential to increase latency in the event that the location of the bucket is physically far from the location from which the request was invoked.

but AFAIU this says "it will work slow in the cases when it would just fail quickly".

Reply:
It incurs an extra round trip to S3 to resolve the bucket's location (region).

Reply:
Ah, so this is not "fallback in case of error" but rather "flexible location apriori". @avirtuos, thanks.
Interesting whether this is real latency hit, though, since javadoc suggests this information might be cached (I'd really prefer it the javadoc was more definitive)

Reply:
@findepi please see https://github.com/aws/aws-sdk-java/issues/1451, there are reasons why this is `false` by default.